sys/include/net/gnrc/tcp/tcb.h

```diff
@@ -88,8 +88,10 @@ typedef struct _transmission_control_block {
     int32_t srtt;          /**< Smoothed round trip time */
     int32_t rto;           /**< Retransmission timeout duration */
     uint8_t retries;       /**< Number of retransmissions */
-    xtimer_t tim_tout;     /**< Timer struct for timeouts */
-    msg_t msg_tout;        /**< Message, sent on timeouts */
+    xtimer_t timer_retransmit; /**< Retransmission timer */
+    xtimer_t timer_misc;       /**< General purpose timer */
+    msg_t msg_retransmit;      /**< Retransmission timer message */
+    msg_t msg_misc;            /**< General purpose timer message */
```

Comment:
Do you thought about `evtimer_msg` instead? This way you only need one memory-intensive `xtimer` for all of TCP (+ it might make migration to `ztimer` easier, see #13661).

Reply:
I didn't know that evtimer_msg exists. I'll take a look to this.

Reply:
You can have a look at `gnrc_ipv6_nib` and `gnrc_rpl` for usage examples.

Reply:
After playing a little bit with evtimer, I am in favor of replacing the xtimer with evtimer within gnrc_tcp. 

Having that said, I would move to evtimer in a seperate PR because it would affect a PR that is currently in review (#14222). 

In addition, evtimer_msg it not sufficient to replace xtimer in gnrc_tcp because messages are sent threads and mboxes during gnrc_tcp operation. Therefore I need to extend evtimer, to be able to send messages to threads and mboxes before I can introduce the change.